pipeline/splicing/calcVarPriors.py

```diff
@@ -667,15 +686,305 @@ def getRefAltScores(refSeq, altSeq, donor=False):
                                "zScore": altZScore}}
     return scoreDict
 
+def getMaxMaxEntScanScoreSlidingWindowSNS(variant):
+    '''
+    Given a variant, determines the maximum alt MaxEntScan score in a 9 bp sliding window
+       with the variant in each position (1-9)
+    Returns a dictionary containing the ref and alt MaxEntScan score and z-score and position of variant for the highest scoring window
+    Dictionary also containing value "inFirstThree" that has value either True or False
+       If inFirstThree = True, then variant is in firt 3 bp of highest scoring sliding window
+       If inFirstThree = False, then variant is in last 6 bp of highest scoring sliding window 
+    '''
+    varGenPos = int(variant["Pos"])
+    varStrand = getVarStrand(variant)
+    # use +- 8 to get 17 bp region so that have sequence for each 9 bp window with variant in positions 1-9
+    # minus strand and plus strand are opposite for +- 8 to preserve sequence returned by getRefAltSeqs
+    if varStrand == "-":
+        regionStart = varGenPos + 8
+        regionEnd = varGenPos - 8
+    else:
+        regionStart = varGenPos - 8
+        regionEnd = varGenPos + 8
+    refAltSeqs = getRefAltSeqs(variant, regionStart, regionEnd)
+    refSeq = refAltSeqs["refSeq"]
+    altSeq = refAltSeqs["altSeq"]
+    varPos = 9
+    windowStart = 0
+    windowEnd = 9
+    windowSeqs = {}
+    windowScores = {}
+    windowAltMaxEntScanScores = {}
+    while windowStart < 9:
+        refWindowSeq = refSeq[windowStart:windowEnd]
+        altWindowSeq = altSeq[windowStart:windowEnd]
+        windowSeqs[varPos] = {"refSeq": refWindowSeq,
+                              "altSeq": altWindowSeq}
+        refAltWindowScores = getRefAltScores(refWindowSeq, altWindowSeq, donor=True)
+        windowScores[varPos] = {"refMaxEntScanScore": refAltWindowScores["refScores"]["maxEntScanScore"],
+                                "refZScore": refAltWindowScores["refScores"]["zScore"],
+                                "altMaxEntScanScore": refAltWindowScores["altScores"]["maxEntScanScore"],
+                                "altZScore": refAltWindowScores["altScores"]["zScore"]}
+        windowAltMaxEntScanScores[varPos] = refAltWindowScores["altScores"]["maxEntScanScore"]
+        varPos -= 1
+        windowStart += 1
+        windowEnd += 1
+        
+    # to get tuple containing sequence with variant position with maximum alt MaxEntScan score
+    maxAltWindowScore = max(windowAltMaxEntScanScores.items(), key=lambda k: k[1])
+    maxVarPosition = maxAltWindowScore[0]
+    maxScores = windowScores[maxVarPosition]
+
+    # determines if variant is in the first three bases of the 9 bp donor sequence
```

Comment:
Again, let's make this a parameter, to minimize the pain when and if we move to a slightly different definition of the critical splice site residues.